### PR TITLE
feat: support explicit segment colors

### PR DESCRIPTION
**Pull request by hubbardp**
In our experience at Janelia, it is helpful for proofreading tools to have explicit control over the colors of bodies in the segmentation.  This code adds that capability to Neuroglancer.  The display state of a layer now can include a map like the following, making one body purple and another green:

"segmentColors": { "193125660": "#ff00ff", "206326381": “#00ff00” }

The implementation uses a new Uint64Map class that is quite similar to the Uint64Set class.  I added some unit tests for both of those classes.

For slice rendering, there is a new SegmentStatedColorShaderManager class that uses a HashMapShaderManager and adds shader code to find a segment’s explicit color if present.  For mesh rendering, the getObjectColor function now checks for the segment’s explicit color.

### CODE REVIEW

**Comment by hubbardp** [comment]
@googlebot I signed it!

File: src/neuroglancer/segmentation_user_layer.ts [hunk 1]
```diff
@@ -157,6 +161,21 @@ export class SegmentationUserLayer extends Base {
       this.specificationChanged.dispatch();
     });
 
+    verifyObjectProperty(specification, SEGMENT_STATED_COLORS_JSON_KEY, y => {
+      if (y !== undefined) {
+        let {segmentEquivalences} = this.displayState;
+        // Parse the color as a 32-bit integer here, because the Uint64.parseString()
+        // function does not seem to handle strings of the form '0xRRGGBB'.
+        // And support a CSS color "#RRGGBB" by turning into "0xRRGGBB".
+        let result = verifyObjectAsMap(y, x => parseInt(String(x).replace(/^#/, '0x'), 0));
```
Comment (jbms) [inline src/neuroglancer/segmentation_user_layer.ts]: You can use parseRGBColorSpecification or parseRGBAColorSpecification in util/color.ts to handle arbitrary CSS color specifications.  Handling alpha might be a bit more work but would probably be quite useful -- but I think that would be reasonable to put off if you don't want to do it now.

**Comment by jbms** [comment]
Overall this looks great!  The only change I'd like to see is with the color parsing.

File: src/neuroglancer/segmentation_user_layer.ts (hunk 1 above)
Comment (hubbardp) [reply src/neuroglancer/segmentation_user_layer.ts]: Thanks for the suggestion.  I just pushed an update using parseRGBColorSpecification.  I think I will put off looking into alpha handling.

**Comment by jbms** [comment]
Thanks!

**Pull request merged**